tests/test_core.py

```diff
@@ -57,12 +59,78 @@ class test_Variable(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different bounds."""
-        raise NotImplementedError()
+        v = relentless.core.Variable(value=1.0, const=True)
+        self.assertAlmostEqual(v.value, 1.0)
+        self.assertEqual(v.const, True)
+
+        v = relentless.core.Variable(value=1.2, low=0.0, high=2.0)
+        self.assertAlmostEqual(v.value, 1.2)
+        self.assertAlmostEqual(v.low, 0.0)
+        self.assertAlmostEqual(v.high, 2.0)
+
+        v = relentless.core.Variable(value=-1.2, low=0.5)
+        self.assertAlmostEqual(v.value, 0.5)
+        self.assertAlmostEqual(v.low, 0.5)
+
+        v = relentless.core.Variable(value=2.2, high=2.0)
+        self.assertAlmostEqual(v.value, 2.0)
+        self.assertAlmostEqual(v.high, 2.0)
+
+        with self.assertRaises(ValueError):
+            v = relentless.core.Variable(value=4)
 
     def test_clamp(self):
         """Test methods for clamping values with bounds."""
-        raise NotImplementedError()
+        v = relentless.core.Variable(value=0.0, low=2.0)
+
+        x = v.clamp(1.0)
```

Comment:
If you want to make your tests a little more readable, you can catch both values in variables like:
```py
val,state = v.clamp(1.0)
```